.test-infra/jenkins/job_PerformanceTests_BigQueryIO_Java.groovy

```diff
@@ -16,102 +16,93 @@
-                numWorkers            : 5,
-                autoscalingAlgorithm  : 'NONE',  // Disable autoscale the worker pool.
+def testJobs = [
+        [
+                title        : 'BigQueryIO Streaming Performance Test Java 10 GB',
+                triggerPhrase: 'Run BigQueryIO Streaming Performance Test Java',
+                jobName      : 'beam_BiqQueryIO_Streaming_Performance_Test_Java',
+                itClass      : 'org.apache.beam.sdk.bigqueryioperftests.BigQueryIOIT',
+                jobProperties: [
+                        project               : 'apache-beam-testing',
+                        tempLocation          : 'gs://temp-storage-for-perf-tests/loadtests',
+                        tempRoot              : 'gs://temp-storage-for-perf-tests/loadtests',
+                        writeMethod           : 'STREAMING_INSERTS',
+                        testBigQueryDataset   : 'beam_performance',
+                        testBigQueryTable     : 'bqio_write_10GB_java',
+                        metricsBigQueryDataset: 'beam_performance',
+                        metricsBigQueryTable  : 'bqio_10GB_results_java_stream',
+                        sourceOptions         : """
+                                            {
+                                              \\\\"numRecords\\\\": \\\\"10485760\\\\",
+                                              \\\\"keySizeBytes\\\\": \\\\"1\\\\",
+                                              \\\\"valueSizeBytes\\\\": \\\\"1024\\\\"
+                                            }
+                                       """.trim().replaceAll("\\s", ""),
+                        runner                : 'DataflowRunner',
+                        maxNumWorkers         : '5',
+                        numWorkers            : '5',
+                        autoscalingAlgorithm  : 'NONE',  // Disable autoscale the worker pool.
+                ]
+        ],
+        [
+                title        : 'BigQueryIO Batch Performance Test Java 10 GB',
+                triggerPhrase: 'Run BigQueryIO Batch Performance Test Java',
+                jobName      : 'beam_BiqQueryIO_Batch_Performance_Test_Java',
+                itClass      : 'org.apache.beam.sdk.bigqueryioperftests.BigQueryIOIT',
+                jobProperties: [
+                        project               : 'apache-beam-testing',
+                        tempLocation          : 'gs://temp-storage-for-perf-tests/loadtests',
+                        tempRoot              : 'gs://temp-storage-for-perf-tests/loadtests',
+                        writeMethod           : 'FILE_LOADS',
+                        testBigQueryDataset   : 'beam_performance',
+                        testBigQueryTable     : 'bqio_write_10GB_java',
+                        metricsBigQueryDataset: 'beam_performance',
+                        metricsBigQueryTable  : 'bqio_10GB_results_java_batch',
+                        sourceOptions         : """
```

Comment:
I wonder why this required such complicated escaping. 😮 If I'm not wrong this could be simplified. Have you tried changing \' to \" in line 102? 

`switches("-DintegrationTestPipelineOptions=\'${common.joinPipelineOptions(testJob.jobProperties)}\'")`
to 
`switches("-DintegrationTestPipelineOptions=\"${common.joinPipelineOptions(testJob.jobProperties)}\"")`

I think these two can be connected and cause weird Jenkins behavior (a wild guess).

Reply:
I'll check if this works. Thanks for the suggestion, I'll get back to you with my test results :)

Reply:
I tried your suggestion and several other options. Sadly, nothing worked. This kind of escaping is necessary, because I used the `enableJavaPerformanceTesting` function from `BeamModulePlugin` - it makes the test expect pipeline options in JSON formatted array. Next, the options are parsed to an actual array, expanded with proper Dataflow-related parameters and written back into a string.

Afterwards, in the Jenkins worker, the options are passed as a bash argument to a command. At this step the quotes inside the options (sourceOptions are a JSON-formatted object nested inside a JSON-formatted array) still need to be escaped for the whole string to be parsed and evaluated properly.

At this point we would either need to modify option parsing inside the `BeamModulePlugin.enableJavaPerformanceTesting` to automatically add the escapes or just leave the escapes as they are right now.

@lgajowy WDYT? What should be the way to go?

Reply:
I see. If this is the case, let's encapsulate the escaping logic in a method to reuse it in multiple places and it will be fine IMO. Thanks for investigating this thoroughly.